coalib/io/FileFactory.py

```diff
@@ -0,0 +1,120 @@
+import linecache
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('_filename', '_stamp')
+class FileFactory:
+    """
+    The FileFactory is a wrapper object to replace the actual file contents
+    from the file-dict. The object contains the following information about
+    the file:
+
+        * The filename (absolute file path).
+        * The parent directory the file is located in.
+        * The modification timestamp of the file.
+        * The file in the form of a string.
+        * The file in the form of a list.
+        * The file in binary format.
+
+    The equality of the object is checked just based on its filename and
+    timestamp.
+
+    To initialize a FileFactory object for a file:
+
+    >>> fp = FileFactory("./tests/io/FileFactoryTestFiles/test.txt")
+
+    Now we can get the filename:
+
+    >>> fp.name
+    './tests/io/FileFactoryTestFiles/test.txt'
+
+    Make the object into an iterable when required
+
+    >>> fp.lines()
+    >>> for line in fp:
+    ...     print(line)
+    This is a test file.
+    <BLANKLINE>
+
+    Retrieve a single line:
+
+    >>> fp.line(1)
+    'This is a test file.\\n'
+
+    Get all the lines in the file as a list:
+
+    >>> fp.lines
+    ['This is a test file.\\n']
+    """
+
+    def __init__(self, filename):
+        """
+        :param filename:
+            The absolute filepath.
+        """
+        self._filename = filename
+        self._parent = os.path.abspath(os.path.dirname(self._filename))
+        self._stamp = os.path.getmtime(self._filename)
+
+    def line(self, line):
+        """
+        :param line:
+            The line number from which the string is to be retrieved.
+        :return:
+            The retrieved string at the given line number from the file.
+        """
+        return linecache.getline(self._filename, line)
+
+    def lines(self):
+        """
+        To get all the lines in the file in form of a list.
+        """
+        self.lines = linecache.getlines(self._filename)
```

Comment:
you don't return something ;)

Reply:
Ah wait, now I understand. You try to cache things here. However, this is highly inconsistent because the first time you have to call it (without a return value even), and next time you can't call it and must access it like a field. Please create consistent API ;)